src/utils/ReentrancyGuard.sol

```diff
@@ -8,7 +8,7 @@ abstract contract ReentrancyGuard {
     uint256 private locked = 1;
 
     modifier nonReentrant() virtual {
-        require(locked == 1, "REENTRANCY");
+        require(locked < 2, "REENTRANCY");
```

Comment:
Lets maybe add a comment about why we do this!

Reply:
Added some comments...

Reply:
OCD Nit: Add a space after the first `//`.